nft_ingester/src/consistency_bg_job.rs

```diff
@@ -0,0 +1,743 @@
+        .unwrap_or(false)
+    {
+        tokio::time::sleep(Duration::from_secs(10)).await;
+    }
+
+    let grand_epoch = grand_epoch_of_epoch(epoch);
+
+    let mut missing_bbgm_changes: HashMap<BbgmChangeRecord, HashSet<usize>> = HashMap::new();
+    let trusted_peers = peers_provider.list_trusted_peers().await;
+
+    for (peer_ind, trusted_peer) in trusted_peers.iter().enumerate() {
+        tracing::info!("Exchanging bubblegum changes for epoch={epoch} with peer={trusted_peer}");
+        let Ok(client) = BbgmConsistencyApiClientImpl::new(
+            trusted_peer,
+            Some(metrics.bbgm_consistency_grpc_client.clone()),
+        )
+        .await
+        .map(Arc::new) else {
+            tracing::warn!("Cannot connect to peer={trusted_peer}");
+            continue;
+        };
+
+        let changes_we_miss = compare_bbgm_with_peer(
+            grand_epoch,
+            storage.as_ref(),
+            client.as_ref(),
+            metrics.as_ref(),
+        )
+        .await;
+
+        metrics
+            .found_missing_bubblegums
+            .set(changes_we_miss.len() as i64);
+
+        for change in changes_we_miss {
+            match missing_bbgm_changes.get_mut(&change) {
+                Some(peers_have_change) => {
+                    peers_have_change.insert(peer_ind);
+                }
+                None => {
+                    missing_bbgm_changes.insert(change, HashSet::from([peer_ind]));
+                }
+            };
+        }
+    }
+    handle_missing_bbgm_changes(
+        missing_bbgm_changes,
+        trusted_peers,
+        force_reingestable_slot_processor,
+    )
+    .await
+}
+
+/// For given grand epoch, compares our bubblegum changes
+/// (by comparing, first, grand epoch checksums and then epoch checksums)
+/// with corresponding bubblegum changes of peer,
+/// and returns changes that are missing on our side.
+///
+/// ## Args:
+/// * grand_epoch - a grand epoch the exchange is performed for
+/// * we - local storage of bubblegum changes (rocksdb storage)
+/// * peer - GRPC client for peer
+pub async fn compare_bbgm_with_peer(
+    grand_epoch: u16,
+    we: &impl BbgmChecksumServiceApi,
+    peer: &impl BbgmChecksumServiceApi,
+    metrics: &Peer2PeerConsistencyMetricsConfig,
+) -> Vec<BbgmChangeRecord> {
+    let mut result = Vec::new();
+    let Ok(peer_ge_chksms) = peer
+        .list_grand_epoch_checksums(grand_epoch, None, None)
+        .await
+    else {
+        return Vec::new();
+    };
+
+    let start = Instant::now();
+    let my_ge_chksms = match we.list_grand_epoch_checksums(grand_epoch, None, None).await {
+        Ok(v) => v,
+        Err(e) => {
+            tracing::error!("Error reading grand epochs from DB: {}", e.to_string());
+            return Vec::new();
+        }
+    };
+    metrics
+        .db_bubblegum_get_grand_epochs_latency
+        .observe(start.elapsed().as_secs_f64());
+
+    let ge_cmp_res = cmp(&my_ge_chksms, &peer_ge_chksms);
```

Comment:
Maybe we need to add some config for understanding which trees we are indexing. Because for now if i understood right we may mark as missed trees that we do not want to index